test/unit/Site.test.js

```diff
@@ -0,0 +1,323 @@
+const Site = require('../../lib/Site');
+const path = require('path');
+const fs = require('fs-extra-promise');
+const ghpages = require('gh-pages');
+const cloneDeep = require('lodash/cloneDeep');
+
+const {
+  INDEX_MD_DEFAULT,
+  PAGE_EJS,
+  SITE_JSON_DEFAULT,
+  USER_VARIABLES_DEFAULT,
+} = require('./utils/data');
+
+jest.mock('fs');
+jest.mock('walk-sync');
+jest.mock('../../lib/Page');
+jest.mock('gh-pages');
+
+
+afterEach(() => fs.vol.reset());
+
+test('Site Generate builds the correct amount of assets', async () => {
+  const json = {
+    'lib/template/page.ejs': PAGE_EJS,
+    'inner/site.json': SITE_JSON_DEFAULT,
+
+    'asset/css/bootstrap.min.css': '',
+    'asset/css/github.min.css': '',
+    'asset/css/markbind.css': '',
+
+    'asset/js/setup.js': '',
+    'asset/js/vue.min.js': '',
+    'asset/js/vue-strap.min.js': '',
+
+    'asset/fonts/glyphicons-halflings-regular.eot': '',
+    'asset/fonts/glyphicons-halflings-regular.svg': '',
+    'asset/fonts/glyphicons-halflings-regular.ttf': '',
+    'asset/fonts/glyphicons-halflings-regular.woff': '',
+    'asset/fonts/glyphicons-halflings-regular.woff2': '',
+  };
+  fs.vol.fromJSON(json, '');
+  const site = new Site('inner/', 'inner/_site');
+  await site.generate();
+  const paths = Object.keys(fs.vol.toJSON());
+  const originalNumFiles = 13;
+  const expectedNumBuilt = 11;
+  expect(paths.length).toEqual(originalNumFiles + expectedNumBuilt);
+
+  // site
+  expect(fs.existsSync(path.resolve('inner/_site'))).toEqual(true);
+
+  // markbind
+  expect(fs.existsSync(path.resolve('inner/_site/markbind'))).toEqual(true);
+
+  // css
+  expect(fs.existsSync(path.resolve('inner/_site/markbind/css'))).toEqual(true);
+  expect(fs.existsSync(path.resolve('inner/_site/markbind/css/bootstrap.min.css'))).toEqual(true);
+  expect(fs.existsSync(path.resolve('inner/_site/markbind/css/github.min.css'))).toEqual(true);
+  expect(fs.existsSync(path.resolve('inner/_site/markbind/css/markbind.css'))).toEqual(true);
+
+  // js
+  expect(fs.existsSync(path.resolve('inner/_site/markbind/js'))).toEqual(true);
+  expect(fs.existsSync(path.resolve('inner/_site/markbind/js/setup.js'))).toEqual(true);
+  expect(fs.existsSync(path.resolve('inner/_site/markbind/js/vue.min.js'))).toEqual(true);
+  expect(fs.existsSync(path.resolve('inner/_site/markbind/js/vue-strap.min.js'))).toEqual(true);
+
+  // fonts
+  expect(fs.existsSync(path.resolve('inner/_site/markbind/fonts/glyphicons-halflings-regular.eot')))
```

Comment:
For consistency with the above tests, you should add a check for the directory `inner/_site/markbind/fonts`.

Actually, could it be removed? The check for the existence of the file implicitly checks for the existence of the directory.

Reply:
Actually I think it's ok not checking the font files... they won't block the site being rendering in the browser anyway :P